Scenes/Entities/Crawler/crawler.gd

```diff
@@ -0,0 +1,59 @@
+extends Node2D
+class_name EnemyNode
+
+# --- Exported Properties ---
+@export var actions: int = 2                ## Number of actions the enemy can take in a single turn.
+@export var speed: float = 150.0            ## Movement speed of the enemy, slightly slower than the player.
+
+# --- Public Properties ---
+var current_cell: Vector2i                  ## Current cell position of the enemy in the grid.
+var target_cell: Vector2                    ## Target cell position for movement.
+var path: Array = []                        ## Array to hold the calculated path for movement.
+var path_index: int = 0                     ## Current index in the path array.
+var move: bool = false                      ## Flag indicating if the enemy is currently performing an action.
+
+# --- Private Properties ---
+@onready var _actions: int = actions
```

Comment:
Would probably rename this to _current_actions. Also probably should be included in Actor base class.